parsers/IN_MH.py

```diff
@@ -0,0 +1,386 @@
+    },
+    #RE TTL
+    'TTL' : {
+            'label' : (597,1035,663,1056),
+            'value' : (783,1032,845,1056)
+    },
+    'TTL (IPP/CPP+RE)' : {
+            'label' : (594,1072,765,1098),
+            'value' : (786,1068,847,1093)
+    },
+    'SOLAR TTL' : {
+            'label' : (592,577,715,605),
+            'value' : (772,583,814,602)
+    },
+    'PIONEER' : {
+            'label' : (592,910,694,929),
+            'value' : (794,910,844,927)
+    }
+}
+
+generation_map = {
+    'biomass' : {
+        'add': ['COGEN'],
+        'subtract': []
+    },
+    'coal' : {
+        'add': ['THERMAL',
+                'TTL (IPP/CPP+RE)',
+                'TPC THM.',
+                'CS GEN. TTL.',
+                'AEML GEN.',
+                'SOLAR TTL'],
+        'subtract': ['TTL',
+                     'PIONEER',
+                     'SSP',
+                     'RGPPL',
+                     'TARPR PH-I',
+                     'TARPR PH-II',
+                     'KK’ PARA',
+                     'GANDHAR']
+    },
+    'gas': {
+        'add': ['GAS',
+                'PIONEER',
+                'GANDHAR',
+                'RGPPL'],
+        'subtract': []
+    },
+    'hydro' : {
+        'add': ['HYDRO',
+                'TPC HYD.',
+                'SSP'],
+        'subtract': []
+    },
+    'nuclear' : {
+        'add': ['TARPR PH-I',
+                'TARPR PH-II',
+                'KK’ PARA'],
+        'subtract': []
+    },
+    'solar' : {
+        'add': ['MS SOLAR'],
+        'subtract': []
+    },
+    'wind' : {
+        'add': ['MS WIND'],
+        'subtract': []
+    },
+    'unknown' : {
+        'add': ['OTHR+SMHYD'],
+        'subtract': []
+    }
+
+}
+
+#list of values that belong to Central State production
+CS = ['SSP',
+     'RGPPL',
+     'TARPR PH-I',
+     'TARPR PH-II',
+     'KK’ PARA',
+     'GANDHAR',
+     'CS GEN. TTL.']
+
+#converts image into a black and white
+def RGBtoBW(pil_image):
+    image = cv2.cvtColor(np.array(pil_image), cv2.COLOR_RGB2GRAY)
+    image = cv2.threshold(image, 0, 255,
+                         cv2.THRESH_BINARY | cv2.THRESH_OTSU)[1]
+    return Image.fromarray(image)
+
+#performs text recognition on given location and source image
+def recognize(location, source, lang):
+    img = source.crop(location)
+    img = RGBtoBW(img)
+    img = ImageOps.invert(img)
+    text = pytesseract.image_to_string(img, lang=lang, config='--psm 7')
+
+    return text, img
+
+# ignores highly indistinguishable characters
+def areEqual(str1,str2):
+    chars = {
+        '{':'(',
+        '}': ')',
+        'T': 'L',
+        'I': 'L',
+        ' ': ''
+    }
+    for key, value in chars.items():
+        str1 = str1.replace(key,value)
+        str2 = str2.replace(key,value)
+    return str1.lower() == str2.lower()
+
+
+def fetch_consumption(zone_key='GE', session=None, target_datetime: datetime.datetime = None,
+                     logger: logging.Logger = None):
```

Comment:
This should be called `fetch_production`, the other keys need to be `target_datetime=None` and `logger=logging.getLogger(__name__)`

Reply:
I'm not so sure about fetch_production.. The output is of this function is the total state demand, ie consumption. Getting the production numbers is a different story since a lot of plants are outside of Maharashtra, see the list of plants in https://github.com/tmrowco/electricitymap-contrib/issues/1443#issue-330342377

Reply:
Maybe I'm slightly confused since the returned json contains a production key.

Reply:
Oh, right, I'll change that!